### PR TITLE
Set default policy for pod exec, port-forward, and proxy

### PR DESCRIPTION
**Pull request by csrwng**

### CODE REVIEW

**Comment by csrwng** [comment]
@smarterclayton @deads2k - this is a WIP proposal for what the default permissions should be for pod exec/portforward/proxy

File: pkg/authorization/api/types.go [hunk 1]
```diff
@@ -67,6 +69,7 @@ var (
 		KubeExposedGroupName:   {"pods", "replicationcontrollers", "services", "endpoints", "pods/log"},
 		KubeAllGroupName:       {KubeInternalsGroupName, KubeExposedGroupName, QuotaGroupName},
 		KubeStatusGroupName:    {"pods/status", "resourcequotas/status", "namespaces/status"},
+		KubePodAdminGroupName:  {"pod/exec", "pod/proxy", "pod/portforward"},
```
Comment (smarterclayton) [inline pkg/authorization/api/types.go]: Logs?

Comment (smarterclayton) [inline pkg/authorization/api/types.go]: Hrm... port forward seems too aggressive to be admin only, as does proxy.  Exec is definitely distinct

Comment (csrwng) [inline pkg/authorization/api/types.go]: Notwithstanding the problem of privileged containers, can we say that in general, if you have read access to the pod, you can do proxy and port forward, and if you have write/update access you can have exec?

Comment (smarterclayton) [inline pkg/authorization/api/types.go]: Yes.  We eventually want to more strongly control exec but we can do that later.

----- Original Message -----

**Comment by brenton** [comment]
Where do we stand on this?

**Comment by deads2k** [comment]
> Where do we stand on this?

Based on discussion earlier this week, I think the roles need reworking to look more like this:
 1. there should be a separate `pod-execer` role.
 1. editors should not be able to exec, proxy, and portforward by default
 1. admins probably can exec, proxy, and portforward by default
 1. there should be a separate `pod-proxy` role for proxy and portforward

@csrwng is that what you took away as well?  If so, you or me?

Also, see https://github.com/GoogleCloudPlatform/kubernetes/issues/8112 for dealing with the weirdness of using `get` for an `exec`.

**Comment by smarterclayton** [comment]
----- Original Message -----
> > Where do we stand on this?
> 
> Based on discussion earlier this week, I think the roles need reworking to
> look more like this:
>  1. there should be a separate `pod-execer` role.
>  1. editors should not be able to exec, proxy, and portforward by default

What?  They should.  

>  1. admins probably can exec, proxy, and portforward by default
>  1. there should be a separate `pod-proxy` role for proxy and port forward

?

> 
> @csrwng is that what you took away as well?  If so, you or me?
> 
> Also, see https://github.com/GoogleCloudPlatform/kubernetes/issues/8112 for
> dealing with the weirdness of using `get` for an `exec`.
> 

We have 3 roles.  Reader, Editor, Project Admin.

**Comment by deads2k** [comment]
> We have 3 roles.  Reader, Editor, Project Admin.

Ok, then add it to editors and this can merge.

**Comment by csrwng** [comment]
@deads2k I removed the group name so I could split the verbs allowed on each resource.

**Comment by deads2k** [comment]
lgtm [merge]

**Pull request merged**